src/Analyser/MutatingScope.php

```diff
@@ -1978,6 +1978,9 @@ private function resolveType(Expr $node): Type
 				if ($resolvedConstantName === '__COMPILER_HALT_OFFSET__') {
 					return new IntegerType();
 				}
+				if ($resolvedConstantName === 'PHP_INT_MAX') {
+					return new IntegerRangeType(1, null);
```

Comment:
Hi, the comment in IntegerRangeType's constructor says: "this constructor can be made private when PHP 7.2 is the minimum"

Everyone should use `IntegerRangeType::fromInterval()` instead.

Reply:
Out-of-scope: but wouldn't it make sense to add `@deprecated` to the constructor then?